drivers/ethernet/eth_nxp_enet.c

```diff
@@ -509,19 +521,44 @@ static void nxp_enet_phy_cb(const struct device *phy,
 {
 	const struct device *dev = eth_dev;
 	struct nxp_enet_mac_data *data = dev->data;
+	const struct nxp_enet_mac_config *config = dev->config;
+	enet_mii_speed_t speed;
+	enet_mii_duplex_t duplex;
+
+	if (state->is_up) {
+#ifdef ENET_HAS_AVB
+		if (PHY_LINK_IS_SPEED_1000M(state->speed)) {
+			speed = kENET_MiiSpeed1000M;
+		} else if (PHY_LINK_IS_SPEED_100M(state->speed)) {
+#else
+		if (PHY_LINK_IS_SPEED_100M(state->speed)) {
```

Comment:
nit but I think compliance allows you to do

```
} else
#endif
if (...) {
```

Reply:
Checkpatch does not take that (suspect code indent for conditional statements).

Reply:
well, that's fine, but this comment would be irrelevant if using a kconfig, since we can use the IS_ENABLED macro then and avoid so much conditional compilation in the driver (my goal is to avoid preprocessor #ifdefs as much as possible)

Reply:
I agree, best is to use `IS_ENABLED()` or different variants if possible.